contracts/implementations/nodes/SyntheticDevice.sol

```diff
@@ -301,6 +309,56 @@ contract SyntheticDevice is
         _resetInfos(syntheticDeviceNode);
     }
 
+
+    /**
+     * @notice Validates burning of a synthetic device and reset all its attributes
+     * @dev Can only be called by the Synthetic Proxy when a token owner calls the `burn` function
+     * @dev The actual burn takes place on the VehicleId contract
+     * @param tokenId Synthetic Device node id
+     */
+    function validateBurnAndResetNode(uint256 tokenId) external onlyNftProxy {
+        NodesStorage.Storage storage ns = NodesStorage.getStorage();
+
+        MapperStorage.Storage storage ms = MapperStorage.getStorage();
+
+        SyntheticDeviceStorage.Storage storage sds = SyntheticDeviceStorage
+            .getStorage();
+
+        VehicleStorage.Storage storage vs = VehicleStorage
+            .getStorage();
+
+        address sdIdProxyAddress = sds.idProxyAddress;
+        address vehicleIdProxyAddress = vs.idProxyAddress;
+
+        if (!INFT(sdIdProxyAddress).exists(tokenId))
+            revert InvalidNode(vehicleIdProxyAddress, tokenId);
+
+        uint256 vehicleNode = ms.nodeLinks[sdIdProxyAddress][vehicleIdProxyAddress][
+                tokenId];
+        if (vehicleNode == 0 ) revert VehicleNotPaired(vehicleNode);
+
+        address owner = INFT(vehicleIdProxyAddress).ownerOf(tokenId);
+
+        //
```

Comment:
What's this?

Reply:
Probably not needed. Event may need it?

Reply:
I just meant a comment with nothing in it.